pkg/apis/core/validation/validation.go

```diff
@@ -3002,12 +3008,16 @@ func validatePodDNSConfig(dnsConfig *core.PodDNSConfig, dnsPolicy *core.DNSPolic
 			}
 		}
 		// Validate searches.
-		if len(dnsConfig.Searches) > MaxDNSSearchPaths {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("searches"), dnsConfig.Searches, fmt.Sprintf("must not have more than %v search paths", MaxDNSSearchPaths)))
+		maxDNSSearchPaths, maxDNSSearchListChars := MaxDNSSearchPathsLegacy, MaxDNSSearchListCharsLegacy
+		if allowExpandedDNSConfig {
```

Comment:
This is not right for update.  You need to check if the old spec already had long data AND the new spec is the same, in which case you validate using the longer limits.  If the old spec was short and the new spec is long, that's an error.  Also, if the old spec was long and the new spec is long, that's an error.

Kubelet always respects the gate and truncates if needed.